app/Http/Controllers/ApiController.php

```diff
@@ -21,6 +24,24 @@ public function __construct(private readonly Request $request)
         $this->initPermissions();
     }
 
+    public function getFacilityOrFail(): Facility
+    {
+        $permissionObject = $this->getPermissionObject();
+        if ($permissionObject->facility) {
+            return $permissionObject->facility;
+        }
+        throw FatalExceptionFactory::unexpected();
+    }
+
+    public function getUserOrFail(): User
+    {
+        $permissionObject = $this->getPermissionObject();
+        if ($permissionObject->user) {
+            return $permissionObject->user;
+        }
+        throw FatalExceptionFactory::unexpected();
```

Comment:
wydaje się, że wszystko można skrócić w tej metodzie do linijki, podobnie wyżej:
```
        return $this->getPermissionObject()->user ?: throw FatalExceptionFactory::unexpected();
```

Reply:
Nie jestem fanem, żeby przez rzeczy typu:

`warunek && akcja()`, `warunek || akcja()` czy `warunek ??/?: akcja` sterować wywołaniami rzeczy

jak już, to spoko by byly `if (warunek) akcja();`, ale to jest niezgodne z PSR :/